### PR TITLE
Verify executable trust in WASM setup() with deferred TOFU recording

### PR DESCRIPTION
**Pull request by mike-north**
## Summary

`resolve_executable_identity`'s `executable_path` branch embedded the caller-supplied string straight into the token's `exe` claim with no hashing, no TOFU manifest consultation, and no Sigstore — so a WASM/Rust consumer that passed `executablePath` got **no** verification despite asking for it, while `identity::trust::verify_trust` sat unused outside its own tests.

This wires trust verification into `VaultKeeper::setup()`: when an `executablePath` is supplied the executable is hashed and run through the tiered checks (Sigstore → trust-manifest match → TOFU first-encounter), the verified hash is bound into the `exe` claim, and a hash conflicting with a previously approved value fails with `IdentityMismatch`. `skipTrust: true` is unchanged.

## Verify / commit split (the #148 ordering constraint)

The existing `verify_trust` coupled the check and the manifest write in one call — importing that shape into `setup()` would reproduce the TS library's open defect #148 (a TOFU record persisted for an operation that later fails). To avoid it, verification is split into two phases:

- **`verify_trust_pending(host, path, opts)` — verify phase, no side effect.** Computes the hash, consults the manifest, and returns a `PendingTrust { identity, tofu_conflict, reason, approved_hashes, manifest_to_save }`. Nothing is written. A first encounter / Sigstore success stashes the updated manifest in `manifest_to_save`; a registry match or a TOFU conflict stashes `None`.
- **`PendingTrust::commit(host)` — commit phase.** Persists `manifest_to_save` (a no-op when `None`).

`setup()` runs the verify phase (after fail-fast shape validation), mints the token, and **only then** calls `commit()`. A failure anywhere before that leaves the manifest untouched. The old `verify_trust` is retained as an eager verify-and-record convenience wrapper (`verify_trust_pending` + immediate `commit`), preserving its existing callers/tests.

`setup()` (Rust core + the WASM binding) becomes `async` to perform the host-bridged hashing and manifest I/O; a TOFU conflict is mapped to `VaultError::IdentityMismatch` (previous = most-recently-approved hash, mirroring the TS `#resolveExecutableIdentity`).

## Acceptance criteria → tests

**AC1 — executable_path calls verify_trust (host-bridged), mirroring the TS tiers, IdentityMismatch on conflict**
- Rust: `setup_with_executable_path_verifies_and_records_hash` — hashes the executable and binds the verified hash (not the raw path) into `exe`.
- WASM: `setup() with an explicit executablePath verifies it and binds the verified hash`.

**AC2 — ordering: no TOFU write before success (do not import #148)**
- Rust: `verify_trust_pending_defers_manifest_write` — the verify phase leaves the manifest unwritten; only `commit` persists it.
- Rust: `verify_trust_pending_conflict_records_nothing` — a conflict carries `approved_hashes` and writes nothing on commit.
- Rust: `setup_with_unreadable_executable_leaves_manifest_unwritten` — a setup that fails during verification records nothing.
- WASM: `#148 ordering: a setup that fails verification leaves the manifest unwritten`.

**AC3 — skipTrust: true bypass unchanged**
- Rust: `setup_with_skip_trust_binds_dev_identity`.
- WASM: `setup() with skipTrust: true opts out and mints a token bound to the dev identity` (existing, still green).

**AC4 — Rust unit + WASM node coverage: first-encounter TOFU, matching hash passes, conflict → mismatch, failed setup leaves manifest unwritten**
- First encounter records TOFU: Rust `setup_with_executable_path_verifies_and_records_hash`; WASM `first encounter records the executable hash under TOFU`.
- Matching hash passes: Rust `setup_with_executable_path_matching_hash_passes`; WASM `a matching hash on a later setup passes and binds the same hash`.
- Conflict → mismatch: Rust `setup_with_executable_path_conflicting_hash_errors_and_records_nothing`; WASM `a conflicting hash throws IdentityMismatchError and records nothing new`.
- Failed setup leaves manifest unwritten (#148 regression): the ordering tests listed under AC2, plus the conflict tests assert the conflicting hash is never persisted.
- Error mapping: WASM `mapWasmError — identity-mismatch code (issue #166)` pins the boundary → typed `IdentityMismatchError` (with `previousHash`/`currentHash`).

**AC5 — wasm rebuilt/committed, JSDoc carve-outs removed, api-report + docs regenerated, changeset**
- Rebuilt `packages/vaultkeeper-wasm/wasm/` via `wasm-pack build --target nodejs`.
- Removed the "does not itself run TOFU/manifest verification" carve-out from `index.ts`, and refreshed `types.ts` / `README.md` (now `await vault.setup(...)`).
- Added a typed `IdentityMismatchError` to the WASM SDK; regenerated `api-report/wasm.api.md` and `docs/api` (`check:api-docs` green).
- Changeset: `minor` on `@vaultkeeper/wasm` (pre-1.0 behavior change: `setup()` is now `async` and enforces verification).

## API surface change (`@vaultkeeper/wasm`)

```diff
-  setup(secretName: string, secretValue: string, options?: SetupOptions): string;
+  setup(secretName: string, secretValue: string, options?: SetupOptions): Promise<string>;
+
+  // new typed error
+  export class IdentityMismatchError extends VaultError {
+    readonly previousHash?: string;
+    readonly currentHash?: string;
+  }
```

## Conformance

Not expressible as a declarative conformance case: the native `vaultkeeper` CLI has no `setup` command (its subcommands are `store` / `exec` / `config` / …), so the `setup()` executable-trust verification is only reachable through the `@vaultkeeper/wasm` SDK, not through the CLI surface the conformance schema drives. The native CLI's own `exec` trust gate is a separate path and is unchanged (its e2e suite stays green). Precedent for a documented no-conformance rationale: PR #135.

## Gate

`cargo test` (native + trust lib), `cargo clippy -D warnings` (native + `wasm32-unknown-unknown`), `cargo fmt --check`, `pnpm check`, `pnpm test` (163 vitest), the `@vaultkeeper/wasm` node:test suite (60), and `pnpm check:api-docs` all pass.

Refs #166

### CODE REVIEW

File: packages/vaultkeeper-wasm/src/test/sdk.test.ts [hunk 1]
```diff
@@ -73,6 +75,51 @@ async function createTestVault(dir: string): Promise<VaultKeeper> {
...
+ * exist (nothing recorded), letting tests assert the #148 ordering property.
+ */
+async function readApprovedHashes(configDir: string, namespace: string): Promise<string[]> {
+  let raw: string;
+  try {
+    raw = await readFile(join(configDir, 'trust-manifest.json'), 'utf8');
+  } catch {
+    return [];
+  }
+  const parsed: unknown = JSON.parse(raw);
```
[the remark this answers is not in the document]
Comment (mike-north) [reply packages/vaultkeeper-wasm/src/test/sdk.test.ts:115]: Good catch — narrowed the catch to return [] only on ENOENT (missing manifest) and rethrow anything else (EACCES/EISDIR/…), so a real failure can't masquerade as no-entries in the manifest-unwritten assertions.

File: packages/vaultkeeper-wasm/src/index.ts [hunk 2]
```diff
@@ -172,15 +173,21 @@ export class VaultKeeper {
...
+   * `'dev'` token. Inspect the error's `reason` (`'missing-choice'` |
+   * `'conflicting-choice'` | `'legacy-dev-sentinel'`) to distinguish the cases.
+   *
+   * **Executable-trust verification.** When `executablePath` is supplied, the
+   * executable is hashed and run through trust verification (Sigstore →
+   * trust-manifest match → TOFU first-encounter), and the verified hash is bound
+   * into the token's `exe` claim. A hash that conflicts with a previously
+   * approved value throws {@link IdentityMismatchError}. The first-encounter
+   * TOFU record is persisted only after the token has minted, so a failed
+   * `setup()` never leaves a premature trust record behind.
```
[the remark this answers is not in the document]
Comment (mike-north) [reply packages/vaultkeeper-wasm/src/index.ts:190]: Fixed — reworded to "after the token has been minted" and regenerated the wasm api-docs (this JSDoc feeds wasm.vaultkeeper.setup.md).

**Comment by mike-north** [comment]
Full review complete — two independent passes (security trace + tests/parity/conventions), plus my own spot-check of the ordering in `vault.rs`. Approving and merging. This closes the gap #161 deferred, and does it properly:

- **Verification is genuinely wired**: `resolve_executable_identity` runs `verify_trust_pending` (Sigstore → manifest match → TOFU) on the `executable_path` branch; the only other `create_token` site is the rotation re-encrypt path (already-authorized token, correctly out of scope). No silent-skip path remains.
- **The deferred-TOFU design is better than the TS reference implementation.** `PendingTrust`/`commit()` splits verify (side-effect-free) from record (post-mint only) — conflict detection reads pre-existing manifest state and records nothing, and both layers have the failed-setup-leaves-manifest-unwritten regression test. The TS library still has the eager write-before-retrieve bug open as #148; this PR pointedly does not import it.
- **Verification quality worth noting**: the committed `.wasm` was rebuilt from this branch's Rust source and came out **byte-for-byte identical** (matching SHA-256), the Rust hash fixtures were independently recomputed (spec-derived, not gold-mastered), and the suites were actually run: 128/128 Rust, 60/60 WASM, full `pnpm check` + `check:api-docs` clean.
- Carve-out JSDoc removed everywhere it appeared, changeset correctly minor-with-breaking-note (setup() now async), and the optional `previousHash`/`currentHash` on the WASM `IdentityMismatchError` follows this package's established boundary-defense precedent rather than drifting from parity.

One optional test-gap note for a future pass: the TOFU-conflict "most recently approved hash" selection (`.last()`/`.at(-1)`) is only exercised with a single-entry manifest history; a 2+-approval history isn't pinned at either layer.

File: packages/vaultkeeper-wasm/src/errors.ts [hunk 3]
```diff
@@ -308,6 +347,8 @@ export function mapWasmError(thrown: unknown): VaultError {
         return new AccessorConsumedError(message);
       case 'executable-trust-required':
         return new ExecutableTrustRequiredError(message, toTrustRequiredReason(thrown.reason));
+      case 'identity-mismatch':
+        return new IdentityMismatchError(message, thrown.previousHash, thrown.currentHash);
```
[the remark this answers is not in the document]
Comment (mike-north) [reply packages/vaultkeeper-wasm/src/errors.ts]: Fixed — mapWasmError now sanitizes both hashes via typeof === 'string' ? value : undefined before constructing IdentityMismatchError, so a malformed boundary value (e.g. { previousHash: null }) stays honestly undefined per the string | undefined contract (same principle as path/permission). Added a mapWasmError unit test with a malformed shape (null / number) asserting both fields come out undefined.

[the remark this answers is not in the document]
Comment (mike-north) [reply packages/vaultkeeper-wasm/src/errors.ts]: Fully addressed across two commits: the rebased revision sanitized the identity-mismatch hashes, and aa6d974 completes the job for every other unvalidated pass-through (path/permission/code on filesystem+decryption, canRefresh on token-expired) — WasmErrorShape's unvalidated fields are now typed unknown so the compiler forces narrowing through shared optionalString/optionalBoolean helpers. 61/61 wasm tests green including the malformed-shape regression cases.

File: crates/vaultkeeper-core/src/identity/trust.rs [hunk 4]
```diff
@@ -53,38 +107,42 @@ pub async fn verify_trust(
...
         }
     }
 
     // --- Tier 2: Registry (manifest) ---
     if is_trusted(&manifest, namespace, &current_hash) {
-        return Ok(TrustVerificationResult {
+        return Ok(PendingTrust {
             identity: IdentityInfo {
                 hash: current_hash,
                 trust_tier: TrustTier::Tofu,
```
[the remark this answers is not in the document]
Comment (mike-north) [reply crates/vaultkeeper-core/src/identity/trust.rs:129]: Refuted with wire-format evidence — this is naming confusion, not a tier bug. Rust's enum is TrustTier::Tofu = 2 (serde rename "2", crates/vaultkeeper-core/src/types.rs:17-19), so a manifest hit mints tid="2" — exactly what the TS library returns for the same case (trustTier: 2, 'Hash found in trust manifest', packages/vaultkeeper/src/identity/trust.ts:95-101). Neither implementation reads a per-entry tier from the manifest (the manifest doesn't record one); both re-check Sigstore first on every call, so a Sigstore-verifiable executable still gets tier 1. Cross-SDK tid parity holds 1:1.

File: crates/vaultkeeper-core/src/identity/trust.rs [hunk 5]
```diff
@@ -9,31 +9,85 @@
...
+    manifest_to_save: Option<TrustManifest>,
+}
+
+impl PendingTrust {
+    /// Persist the pending TOFU manifest write, if any. A no-op when there is
+    /// nothing to record.
+    ///
+    /// Call this only after the overall operation has otherwise succeeded (e.g.
+    /// the token has minted) so a failure never leaves a premature TOFU record
+    /// behind (issue #148).
```
[the remark this answers is not in the document]
Comment (mike-north) [reply crates/vaultkeeper-core/src/identity/trust.rs:53]: Fixed in 332998d ('has been minted' across trust.rs, wasm_impl.rs, the changeset, and the regenerated wasm glue — binary byte-identical, only the doc line changed).

File: crates/vaultkeeper-wasm/src/wasm_impl.rs [hunk 6]
```diff
@@ -478,7 +486,12 @@ impl WasmVaultKeeper {
     }
 
     /// Create a JWE token encapsulating a secret.
-    pub fn setup(
+    ///
+    /// When `options.executablePath` is supplied, the executable is hashed and
+    /// run through trust verification (Sigstore → trust-manifest match → TOFU
+    /// first-encounter) via the host bridge; a first-encounter TOFU record is
+    /// persisted only after the token has minted (issue #148).
```
[the remark this answers is not in the document]
Comment (mike-north) [reply crates/vaultkeeper-wasm/src/wasm_impl.rs]: Fixed in 332998d.

File: packages/vaultkeeper-wasm/wasm/vaultkeeper_wasm.js [hunk 7]
```diff
@@ -187,32 +187,23 @@ class WasmVaultKeeper {
     }
     /**
      * Create a JWE token encapsulating a secret.
+     *
+     * When `options.executablePath` is supplied, the executable is hashed and
+     * run through trust verification (Sigstore → trust-manifest match → TOFU
+     * first-encounter) via the host bridge; a first-encounter TOFU record is
+     * persisted only after the token has minted (issue #148).
```
[the remark this answers is not in the document]
Comment (mike-north) [reply packages/vaultkeeper-wasm/wasm/vaultkeeper_wasm.js]: Fixed in 332998d — regenerated via wasm-pack from the corrected Rust doc comment rather than hand-editing the generated file.

File: crates/vaultkeeper-wasm/src/wasm_impl.rs (hunk 6 above)
[the remark this answers is not in the document]
Comment (mike-north) [reply crates/vaultkeeper-wasm/src/wasm_impl.rs]: Fixed in 332998d.

File: .changeset/wasm-setup-executable-trust-verification.md [hunk 8]
```diff
@@ -0,0 +1,13 @@
+---
+'@vaultkeeper/wasm': minor
+---
+
+Enforce executable-trust verification in `setup()` when an `executablePath` is supplied.
+
+Previously, passing `executablePath` bound the raw path into the token's `exe` claim with no hashing and no trust-manifest consultation — a caller that explicitly asked for executable trust got none. `setup()` now hashes the executable and runs trust-on-first-use verification (Sigstore → trust-manifest match → TOFU first-encounter) through the host bridge, binding the verified hash into the `exe` claim, matching the pure-TypeScript `vaultkeeper` library's behavior.
+
+- A first encounter records the executable's hash under trust-on-first-use. A later `setup()` with a matching hash passes; a changed hash throws the new `IdentityMismatchError` (carrying `previousHash` / `currentHash`) rather than silently re-approving.
+- The first-encounter manifest write is committed only after the token has minted, so a failed `setup()` never leaves a premature trust record behind.
```
[the remark this answers is not in the document]
Comment (mike-north) [reply .changeset/wasm-setup-executable-trust-verification.md]: Fixed in 332998d.
